compiler/rustc_llvm/build.rs

```diff
@@ -103,8 +103,10 @@ fn main() {
         println!("cargo:rustc-check-cfg=values(llvm_component,\"{}\")", component);
     }
 
-    if tracked_env_var_os("RUST_CHECK").is_some() {
+    if env::var_os("RUST_CHECK").is_some() {
         // If we're just running `check`, there's no need for LLVM to be built.
+        // We only track the RUST_CHECK var if it exists to prevent spurious rebuilds.
+        println!("cargo:rerun-if-env-changed=RUST_CHECK");
```

Comment:
This is pretty suspicious -- this env variable *should* get set by rustbuild only if LLVM hasn't been built and we don't expect to need it (check/clippy run).

> causing it to rerun when checking after building

So if you build, then on a subsequent x.py check, RUST_CHECK shouldn't be set. I think the fault here lies with the logic around here (https://github.com/rust-lang/rust/blob/3830ecaa8db798d2727cbdfa4ddf314ff938f268/src/bootstrap/builder.rs#L1391), not this build script.